src/VisualStudio/Core/Def/Implementation/FindReferences/Entries/DocumentSpanEntry.cs

```diff
@@ -200,6 +201,25 @@ private static Span GetRegionSpanForReference(SourceText sourceText, TextSpan so
                     sourceText.Lines[firstLineNumber].Start,
                     sourceText.Lines[lastLineNumber].End);
             }
+
+            bool ISupportsNavigation.TryNavigateTo(bool isPreview)
+            {
+                // If the document is a source generated document, we need to do the navigation ourselves;
+                // this is because the file path given to the table control isn't a real file path to a file
+                // on disk.
+                if (_excerptResult.Document is SourceGeneratedDocument)
+                {
+                    var workspace = _excerptResult.Document.Project.Solution.Workspace;
+                    var documentNavigationService = workspace.Services.GetService<IDocumentNavigationService>();
+
+                    if (documentNavigationService != null)
+                    {
+                        return documentNavigationService.TryNavigateToSpan(workspace, _excerptResult.Document.Id, _excerptResult.Span);
```

Comment:
Do we also need to pass `options: options.WithChangedOption(NavigationOptions.PreferProvisionalTab, isPreview)` for single clicking results, or is that handled automatically/elsewhere?

Reply:
Oh right I forgot to test this.

Reply:
Fixed, with a dirty hack to work around something that was breaking provisional tabs: https://github.com/dotnet/roslyn/pull/50356/commits/744d50b3ed5bd21af07fc0c09448081f4d7f49bb